Document recovery and backup_disabled account data

uhoreg opened the pull request.

uhoreg committed:
document recovery and backup_disabled account data

uhoreg committed:
add note about Rust implementation

richvdh reviewed (changes requested):
As already discussed elsewhere:

 * I think it might be a bit too verbose.

   I realise (somewhat belatedly) that that's what the existing files in element-meta/spec are doing, but I don't really think it's appropriate to follow the MSC format here. If we wanted to make a proposal to change the spec, complete with problem statement, proposal, and security considerations, then why wouldn't we make it an actual MSC.

   The whole point here is that Element has already decided to introduce these new account data events, and now we just need to record their existence and write down their semantics.

* It kinda feels like we need a top-level doc saying:

   ```
   The Element apps use these non-standard events:

   # Account data events

   - io.element.recovery
   - io.element.xyz
   ```

   ... with a link to the doc for each event

 * also also.... I'm not really sure we need a separate doc for `m.org.matrix.custom.backup_disabled`, given it's documented as part of MSC4287?

   the main thing to have something to link to from the code, and we can do that with the MSC?

uhoreg committed:
simplify documentation, and add summary

uhoreg commented:
I've simplified the docs, and added a top-level doc summarizing the event types.  I kept the documentation for `m.org.matrix.custom.backup_disabled` since it's already written.

richvdh reviewed (approved):
TYVM
